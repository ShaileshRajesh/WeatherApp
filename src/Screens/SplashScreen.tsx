import React from 'react';
import {View, StyleSheet, SafeAreaView} from 'react-native';
import Button from '../Components/Button';
import {AppColors, buttonColors} from '../Utils/Colors';
import {ROUTES, STRINGS} from '../Utils/Constants';
import Image from '../Components/Image';
import Text from '../Components/Text';
import {fontFamily} from '../Utils/Styling';
import useNavigate from '../CustomHooks/useNavigate';
import useAuth from '../CustomHooks/useAuth';

/**
 * @screen
 * Splash screen which is shown intially in the application.
 */

const SplashScreen = () => {
  const navigate = useNavigate();
  const {isUserLoggedIn} = useAuth();

  const onHandlePress = () => {
    if (isUserLoggedIn) {
      navigate(ROUTES.APP_STACK);
    } else {
      navigate(ROUTES.AUTH_STACK);
    }
  };
  return (
    <SafeAreaView style={styles.container}>
      <View style={styles.imageContainer}>
        <Image
          isBackgroundImage
          bgImgsource={require('../../assets/Icons/blueBackground.png')}
          backgroungImgStyle={styles.backgroundImage}>
          <Image
            source={require('../../assets/Icons/droplets.png')}
            imgStyle={styles.droplets}
          />
          <Image
            source={require('../../assets/Icons/walking_in_rain.png')}
            imgStyle={styles.walkingMan}
          />
        </Image>
        <View style={styles.subContainer}>
          <Text textStyle={styles.rainDropText}>{STRINGS.RAINDROPS}</Text>
          <Text textStyle={styles.discription1}>
            {STRINGS.DISCRIPTION1}
            {'\n'}
            <Text textStyle={styles.discription2}>{STRINGS.DISCRIPTION2}</Text>
          </Text>
          <Button
            label={STRINGS.GET_STARTED}
            backgroundColor={buttonColors.blue}
            border={25}
            buttonStyle={styles.buttonContainer}
            textStyle={styles.buttonText}
            onClick={onHandlePress}
          />
        </View>
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    backgroundColor: AppColors.lightBlue,
  },
  buttonContainer: {
    width: '90%',
    marginTop: 25,
    paddingVertical: 8,
  },
  imageContainer: {
    width: 300,
    height: 320,
    marginTop: '5%',
  },
  backgroundImage: {
    height: '100%',
    resizeMode: 'cover',
  },
  droplets: {
    alignSelf: 'center',
    justifyContent: 'center',
    top: '5%',
  },
  walkingMan: {
    alignSelf: 'center',
    position: 'absolute',
    top: '35%',
    left: '5%',
  },
  subContainer: {
    justifyContent: 'center',
    alignItems: 'center',
    marginTop: '45%',
  },
  rainDropText: {
    fontFamily: fontFamily.bold,
    fontSize: 48,
    color: buttonColors.blue,
  },
  discription1: {
    fontFamily: fontFamily.regular,
    fontSize: 18,
    textAlign: 'center',
    color: buttonColors.blue,
  },
  discription2: {
    fontFamily: fontFamily.regular,
    fontSize: 18,
    textAlign: 'center',
    color: buttonColors.blue,
  },
  buttonText: {
    color: AppColors.white,
    fontSize: 12,
    fontFamily: fontFamily.regular,
  },
});
export default SplashScreen;
