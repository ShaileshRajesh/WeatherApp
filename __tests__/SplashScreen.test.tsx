import renderer from 'react-test-renderer';
import SplashScreen from '../src/Screens/SplashScreen';

test('renders splash screen', () => {
  const tree = renderer.create(<SplashScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
