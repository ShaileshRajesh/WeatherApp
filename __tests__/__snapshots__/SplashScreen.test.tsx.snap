// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders splash screen 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#C8D9F9",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "height": 320,
        "marginTop": "5%",
        "width": 300,
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        {
          "height": "100%",
          "resizeMode": "cover",
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../assets/Icons/blueBackground.png",
          }
        }
        style={
          [
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            {
              "height": "100%",
              "width": undefined,
            },
            undefined,
          ]
        }
      />
      <Image
        resizeMode="cover"
        source={
          {
            "testUri": "../../../assets/Icons/droplets.png",
          }
        }
        style={
          {
            "alignSelf": "center",
            "justifyContent": "center",
            "top": "5%",
          }
        }
      />
      <Image
        resizeMode="cover"
        source={
          {
            "testUri": "../../../assets/Icons/walking_in_rain.png",
          }
        }
        style={
          {
            "alignSelf": "center",
            "left": "5%",
            "position": "absolute",
            "top": "35%",
          }
        }
      />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "justifyContent": "center",
          "marginTop": "45%",
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#0F62FE",
              "fontFamily": "AnonymousPro-Bold",
              "fontSize": 48,
            },
          ]
        }
      >
        raindrops
      </Text>
      <Text
        style={
          [
            {
              "color": "#0F62FE",
              "fontFamily": "AnonymousPro-Regular",
              "fontSize": 18,
              "textAlign": "center",
            },
          ]
        }
      >
        Only grab your umbrella when
        

        <Text
          style={
            [
              {
                "color": "#0F62FE",
                "fontFamily": "AnonymousPro-Regular",
                "fontSize": 18,
                "textAlign": "center",
              },
            ]
          }
        >
          you really need it.
        </Text>
      </Text>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#0F62FE",
            "borderRadius": 25,
            "justifyContent": "center",
            "marginTop": 25,
            "opacity": 1,
            "paddingVertical": 8,
            "width": "90%",
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#FFFFFF",
                "textAlign": "center",
              },
              {
                "color": "#FFFFFF",
                "fontFamily": "AnonymousPro-Regular",
                "fontSize": 12,
              },
            ]
          }
        >
          Get started
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
